mlir/include/mlir/Dialect/Utils/IndexingUtils.h

```diff
@@ -243,6 +243,14 @@ SmallVector<int64_t>
 computePermutationVector(int64_t permSize, ArrayRef<int64_t> positions,
                          ArrayRef<int64_t> desiredPositions);
 
+/// Returns a permutation vector that remove the result position in
+/// removePositions from inputPerm.
+///
+/// For example, inputPerm = {2, 4, 0, 1, 3} and removePositions = {1, 2} would
+/// result in a {2, 0, 1} permutation vector.
```

Comment:
Is the result correct? Shouldn't this be `{2, 1, 3}` or am I getting this wrong?

Reply:
Sorry not make this function clear.
This function returns a *new permutation* after removing  *input position* in removePositions.

The removed position is "2", "1" in input pos,  after remove, we have {4, 0, 3}.
To be a valid permutation, returned perm should start from "0", result should be {2, 0, 1}.

Reply:
Thanks! I think we should already have this functionality implemented on `AffineMap`. Would you mind taking a look at the utilities in AffineMap.h? There are some `drop...` methods might get you what you need. You can get an AffineMap from a permutation with: https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/IR/AffineMap.h#L103

Reply:
Thank you for your suggestion, this approach is very interesting (at least my function name is not precise enough, I think dropResults would be better). We can indeed replace the calculation of perm with the calculation of affine map using applyPermutationMap and getPermutationMap. Before making the changes, I would like to discuss some points: This will simplify the calculation of perm, but it will introduce more compilation time -- we have to construct affine maps to reuse the affine map operation functions. Is this a more reasonable approach? If so, all the util functions in the Permutation utils series have corresponding util functions in the affine map. Should we systematically replace them all?

Reply:
Good point. My comment was motivated by the high proliferation of AffineMap and permutation utilities over the years (to the point that sometimes it's a challenge, even for people familiar with the code, to figure out if something exists already). However, I think this adding this one is justified as it's combined with other utilities that work on permutations.